store/storetest/preference_store.go

```diff
@@ -372,17 +373,17 @@ func testPreferenceDeleteCategory(t *testing.T, ss store.Store) {
 	require.Nil(t, err)
 	require.Equal(t, 2, count)
 
-	if prefs := store.Must(ss.Preference().GetAll(userId)).(model.Preferences); len([]model.Preference(prefs)) != 2 {
-		t.Fatal("should've returned 2 preferences")
-	}
+	preferences, err := ss.Preference().GetAll(userId)
+	require.Nil(t, err)
+	assert.NotEqual(t, len(preferences), 2, "should've returned 2 preferences")
```

Comment:
nit: You can use here `assert.Len` (and in all the other assertions where you have used this pattern.

Reply:
From [this](https://godoc.org/github.com/stretchr/testify/assert#Len) i got that i can only assert if the length is equal to the value we want. In this case we wanted to prove the opposite. Although i get your point, but I would have to re-write the test to use `assert.len()` and I thought this would be quicker and would still make sense.  If there is a problem with that i can change it ofc 😄

Reply:
Actually not, the previous code was al the way around if length != 2 then `t.Fatal` so the assertion should be "lenght == 2", if the assertion is not fulfilled, then fail with the provided error.